rs/rosetta-api/src/request_handler.rs

```diff
@@ -457,12 +457,14 @@ impl RosettaRequestHandler {
             hash: Some(msg.block_identifier.hash),
         });
         let mut block = self.get_block(block_id).await?;
-        // TODO: for Rosetta blocks fetch the right transactions
+
         let transaction = match self.rosetta_blocks_mode().await {
             RosettaBlocksMode::Disabled => block.transactions.remove(0),
```

Comment:
Is it not possible to specify the `transaction_identifier` in the case of `RosettaBlocksMode::Disabled` also? In that case, shouldn't we check the tx ID here also, and return `ApiError::InvalidTransactionId` in that case?

Reply:
You can specify the `transaction_identifier` for ICP blocks as well but there is little point to it, as you can just get the same result by using the `/block`endpoint.